public/app/plugins/panel/heatmap-new/ColorScale.tsx

```diff
@@ -10,55 +10,72 @@ type Props = {
   const [colors, setColors] = useState<string[]>([]);
-  const [hover, setHover] = useState<HoverState>({ isShown: false, value: 0 });
-  const [cursor, setCursor] = useState({ clientX: 0, clientY: 0 });
+  const [scaleHover, setScaleHover] = useState<HoverState>({ isShown: false, value: 0 });
+  const [percent, setPercent] = useState<number | null>(null);
+  const [eventOffsetWidth, setEventOffsetWidth] = useState<number>(0);
+
+  const theme = useTheme2();
+  const styles = getStyles(theme, colors);
 
   useEffect(() => {
     setColors(getGradientStops({ colorArray: colorPalette }));
   }, [colorPalette]);
 
-  const theme = useTheme2();
-  const styles = getStyles(theme, colors);
+  const offsetWidth = useMemo(() => eventOffsetWidth, [eventOffsetWidth]);
 
   const onScaleMouseMove = (event: React.MouseEvent<HTMLDivElement>) => {
+    setEventOffsetWidth((event.target as any).offsetWidth as number);
     const divOffset = event.nativeEvent.offsetX;
-    const offsetWidth = (event.target as any).offsetWidth as number;
     const normPercentage = Math.floor((divOffset * 100) / offsetWidth + 1);
     const scaleValue = Math.floor(((max - min) * normPercentage) / 100 + min);
-    setHover({ isShown: true, value: scaleValue });
-    setCursor({ clientX: event.clientX, clientY: event.clientY });
+
+    setScaleHover({ isShown: true, value: scaleValue });
+    setPercent(normPercentage);
   };
 
   const onScaleMouseLeave = () => {
-    setHover({ isShown: false, value: 0 });
+    setScaleHover({ isShown: false, value: 0 });
   };
 
+  useEffect(() => {
+    if (hoverValue != null) {
+      const percent = hoverValue / (max - min);
+      setPercent(percent * 100);
+    }
+  }, [hoverValue, min, max]);
+
   return (
     <div className={styles.scaleWrapper}>
-      <div>
-        <div className={styles.scaleGradient} onMouseMove={onScaleMouseMove} onMouseLeave={onScaleMouseLeave}>
-          {display && hover.isShown && (
-            <VizTooltipContainer position={{ x: cursor.clientX, y: cursor.clientY }} offset={{ x: 10, y: 10 }}>
-              {display(hover.value)}
-            </VizTooltipContainer>
-          )}
-        </div>
-        {display && (
-          <div>
-            <span>{display(min)}</span>
-            <span className={styles.maxDisplay}>{display(max)}</span>
+      <div className={styles.scaleGradient} onMouseMove={onScaleMouseMove} onMouseLeave={onScaleMouseLeave}>
+        {display && (scaleHover.isShown || hoverValue !== undefined) && (
+          <div className={styles.followerContainer}>
+            <div className={styles.follower} style={{ left: percent + '%' }} />
           </div>
         )}
       </div>
+      {display && (
+        <div style={{ position: 'relative' }}>
+          {percent != null && (scaleHover.isShown || hoverValue !== undefined) && (
+            <span style={{ position: 'absolute', left: percent - 2 + '%', paddingTop: '5px' }}>
```

Comment:
^^ move the constant parts of the style to emotion class, but keep the dynamic parts as style